src/reducers-actions/trades.ts

```diff
@@ -1,20 +1,26 @@
 import { Actions } from 'reducers-actions'
 import { Trade } from 'api/exchange/ExchangeApi'
 
-export type ActionTypes = 'APPEND_TRADES'
+export type ActionTypes = 'APPEND_TRADES' | 'UPDATE_BLOCK'
 
 export interface TradesState {
   trades: Trade[]
   lastCheckedBlock?: number
 }
 
+type UpdateBlockActionType = Actions<ActionTypes, Required<Pick<TradesState, 'lastCheckedBlock'>>>
 type ReducerActionType = Actions<ActionTypes, TradesState>
```

Comment:
Need more specificity
```suggestion
type UpdateBlockActionType = Actions<'UPDATE_BLOCK', Required<Pick<TradesState, 'lastCheckedBlock'>>>
type ReducerActionType = Actions<'APPEND_TRADES', TradesState>
```

like in` src/reducers-actions/localTokens.ts`
Otherwise in `switch..case` in reducer TS won't be able to differentiate between actions, and see action as `UpdateBlockActionType | ReducerActionType`

Also rename `ReducerActionType` to `UpdateTradeStateActionType` or something